### ISSUE TITLE
[ROADMAP] Kornia v0.1.5

### ISSUE THREAD

**Issue by edgarriba** [kornia/kornia#312]
This is the roadmap for **kornia v0.1.5** focuses mainly importing in the data augmentation module.

Feel free to volunteer yourself if you are interested in trying out some items(they do not have to be on the list). Please, take a read on the [CONTRIBUTING](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst) notes.

* **Data augmentation** https://github.com/kornia/kornia/issues/298 -> ticket PR https://github.com/kornia/kornia/pull/309
Make differentiable versions of albumentations, torchvision. Please propose and we will assign tasks.
  * [x] BaseAugmentation @kshitij12345 @edgarriba d729d7c4357ca73e4915a42285a0771bca4436ce
  * [x] Denormalize @dvd42 0972e55b221d4d2cedd3c44c72ac55dfe7bd5d50
  * [x] ColorJitter @priba 98b875289cc05e61fc12bcf8775640f57fd4d273
  * [x] RandomHorizontalFlip @dvd42 ba1e89c6d7ff0a38bb171c0cfbd7bdc7726018c8
  * [x] MotionBlur @mronta 63a34dfbbbc2fcf1bcbfe94547ad6c29896058b9
  * [x] RandomVerticalFlip @kshitij f6f1c1164805a8b60a4e5f28d0dc9146360c07fd
  * [x] RandomErasing @mronta 85ef4c0a1e58a023dbae840b233677b37ac32fdc
  * [x] RandomGrayscale @priba 1782eb35bc61600df65c17087c7a03e3aa93d1ee
  * [x] Resize @edgarriba d0f596eee9d49d8a5cdd46d9ced9d02ae80423ab
  * [x] CenterCrop @edgarriba https://github.com/kornia/kornia/pull/409
  * [x] RandomAffine @edgarriba  https://github.com/kornia/kornia/pull/403
  * [x] RandomPerspective @edgarriba https://github.com/kornia/kornia/pull/403
  * [x] RandomRotation @dvd42 https://github.com/kornia/kornia/pull/397
  * [x] RandomResizedCrop @kshitij https://github.com/kornia/kornia/pull/408
  * [ ] Grayscale @dvd42 
    * call rgb_to_grayscale + https://pytorch.org/docs/stable/torchvision/transforms.html#torchvision.transforms.Grayscale

* **Color transforms**
Match missing OpenCV conversion. **Please propose !**
  * [x]  rgb_to_ycbcr @justanhduc a0bafcdc8f959ad1ddcae5b050a8d786e1e66000
  * [x]  ycbcr_to_rgb @justanhduc a0bafcdc8f959ad1ddcae5b050a8d786e1e66000
  * [x] rgb_to_yuv @ananyamukh6 eefca4f443703c133d4276b83d0f09aae85042de
  * [x] yuv_to_rgb @ananyamukh6 eefca4f443703c133d4276b83d0f09aae85042de
  * [x] rgb_to_rgba @ucalyptus @edgarriba e6689bec210c23f385d076e6dee2bcd47272d5de
  * [x] rgba_to_rgb @ucalyptus @edgarriba e6689bec210c23f385d076e6dee2bcd47272d5de
  * [x] bgr_to_bgra @ucalyptus @edgarriba e6689bec210c23f385d076e6dee2bcd47272d5de
  * [x] bgra_to_bgr @ucalyptus @edgarriba e6689bec210c23f385d076e6dee2bcd47272d5de
  * [ ] lab_to_rgb https://github.com/kornia/kornia/issues/239 @HassanAlsamahi 
  * [ ] rgb_to_lab https://github.com/kornia/kornia/issues/239 @HassanAlsamahi 
  * [ ] rgb_to_xyz @priba
  * [ ] xyz_to_rgb @priba
  * [ ] rgb_to_luv @priba
  * [ ] luv_to_rgb @priba

* **GPU Tests**

  * [x] parametrize test functions to accept `torch.device` cpu/cuda @edgarriba @ducha-aiki da793cd48e17eca7e78b042240a9464065611d9d 0fcb85e25f0d37042ff426a2541cfb296b03cd30
  * [ ] setup CI @edgarriba @chadell

* **Framework**

  * [x] Update to PyTorch 1.4.0 @edgarriba https://github.com/kornia/kornia/pull/402

**Comment by ananyamukh6** [comment]
Volunteering for Morphology, some color conversions.

**Comment by kshitij12345** [comment]
@edgarriba I would like to work on `Random{Vertical/Horizontal}Flip`

**Comment by edgarriba** [comment]
@kshitij12345 sure, once we have decided how the api should look like and merge this https://github.com/kornia/kornia/pull/309/files

**Comment by edgarriba** [comment]
@kshitij12345 augmentation module is ready to be populated

**Comment by ucalyptus** [comment]
Would like to work on rgb_to_rgba

**Comment by edgarriba** [comment]
@ucalyptus go ahead

**Comment by justanhduc** [comment]
@edgarriba I already have RGB2 <-> YCbCr. May I work on that?

**Comment by edgarriba** [comment]
@justanhduc yes, please :)

**Comment by mronta** [comment]
@edgarriba I can apply for random erasing

**Comment by edgarriba** [comment]
@mronta sure go ahead. In contrast to torchvision, I propose to extend to generate batched parameters to erase. Besides, I would think in performing this op in a soft way /cc @ducha-aiki

**Comment by mronta** [comment]
@edgarriba is differentiable image resize already achievable through  `torch.nn.functional.interpolate` ?

**Comment by ducha-aiki** [comment]
@mronta yes, but for proper resize you need to blur with appropriate kernel size first.

**Comment by shijianjian** [comment]
For data augmentation tasks, I think it is better to mimic PyTorch style implementation for better reproducibility.

Take the ```ColorJitter``` as an example, the current implementation cannot reproduce the same result  via a function call. Currently, we have:
    1. ```colorJitter``` method, which will generate random params and apply function.
    2. ```ColorJitter``` class, which does nothing but calls the ```colorJitter``` method.

For better reproducibility, I think the PyTorch style is more clear that for ```ColorJitter``` class, it will have a ```get_params``` static method that generates random params with a ```def __call__()``` implementation. Thus, the ```colorJitter``` method will apply the transformations only.

One of the use cases of reproducing the transformations can be for segmentation tasks, that we will wish to apply the exact same affine transformations for both input image and mask.

**Comment by ceroytres** [comment]
Can I help with the remaining color conversions?  @edgarriba

**Comment by edgarriba** [comment]
closing in favor of https://github.com/kornia/kornia/issues/434

**Comment by edgarriba** [comment]
@ceroytres check new roadmap and please pick any item :D
